billboard.py

```diff
@@ -350,6 +431,8 @@ def _parsePage(self, soup):
 
         if soup.select("table"):
             self._parseOldStylePage(soup)
+        elif self.year:
```

Comment:
Should this check come first? For dated (regular) charts, there are "old-style" and "new-style" pages (that have different DOM structures). I'm guessing that distinction doesn't exist for year-end charts?

Reply:
Oh sure, that would make sense to check for `self.year` first. My impression from looking at your code is that the year-end pages are most similar to the "new-style" pages, but with slightly different tags.